From: Armin Novak <armin.novak@thincast.com>
Date: Tue, 8 Mar 2022 15:21:23 +0100
Subject: rfx_process_message verbose error log

(cherry picked from commit c194a68b1bb44af3a180d7bdad678655bd673dc9)
(cherry picked from commit 79207e6700e83ffcbd707985086bb3a6dc06dfb9)
---
 libfreerdp/codec/rfx.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/libfreerdp/codec/rfx.c b/libfreerdp/codec/rfx.c
index 8c65e75..b41c157 100644
--- a/libfreerdp/codec/rfx.c
+++ b/libfreerdp/codec/rfx.c
@@ -1071,7 +1071,10 @@ BOOL rfx_process_message(RFX_CONTEXT* context, const BYTE* data, UINT32 length,
 			UINT8 channelId;
 
 			if (Stream_GetRemainingLength(s) < 2)
+			{
+				WLog_ERR(TAG, "extraBlockLen too small(%" PRIuz ")", Stream_GetRemainingLength(s));
 				return FALSE;
+			}
 
 			extraBlockLen = 2;
 			Stream_Read_UINT8(s, codecId);   /* codecId (1 byte) must be set to 0x01 */
@@ -1212,6 +1215,9 @@ BOOL rfx_process_message(RFX_CONTEXT* context, const BYTE* data, UINT32 length,
 				                        NULL, FREERDP_FLIP_NONE))
 				{
 					region16_uninit(&updateRegion);
+					WLog_ERR(TAG,
+					         "nbUpdateRectx[% " PRIu32 " (%" PRIu32 ")] freerdp_image_copy failed",
+					         j, nbUpdateRects);
 					return FALSE;
 				}
 
@@ -1226,6 +1232,7 @@ BOOL rfx_process_message(RFX_CONTEXT* context, const BYTE* data, UINT32 length,
 		return TRUE;
 	}
 
+	WLog_ERR(TAG, "%s failed", __FUNCTION__);
 	return FALSE;
 }
 
